### PR TITLE
Added clear selected interaction

### PR DESCRIPTION
**Pull request by sebalp100**
### **Added clear selected interaction**

For this milestone we added:
- New event listener for checkboxes (change)
- New event listener for the clear completed button (completed: true/false).
- New method to delete completed tasks (using filter() method)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -33,4 +33,21 @@ list.addEventListener('keypress', (e) => {
...
+  const tasks = JSON.parse(localStorage.getItem('tasks'));
+  checkbox.forEach((checkbox) => {
+    if (checkbox.checked) {
+      tasks[checkbox.id].completed = true;
+    } else {
+      tasks[checkbox.id].completed = false;
+    }
+  });
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+});
```
Comment (prolajumokeoni) [inline src/index.js:53]: - [x] Based on the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m2_interactive_list.md#project-requirements), please use the filter() method to implement a function for the "Clear all completed" button

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @sebalp100 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used 
✔️  Good PR title and Description



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (sebalp100) [reply src/index.js:53]: @prolajumokeoni Hi, I'm using the filter method in the objects.js file. It gets called when the user clicks on the clear all complete button. Not sure if I understand what you mean.

**Comment by sebalp100** [comment]
Added filter() method for the clear completed task button, which gets called from the isChecked method in object.js

> static isChecked() {
    const tasks = JSON.parse(localStorage.getItem('tasks'));
    const checkboxIds = document.querySelectorAll('.checkbox'); // select all checkboxes
    const counter = []; // empty array to store task index
    checkboxIds.forEach((checkbox) => {
      if (checkbox.checked === true) {
        const targetId = checkbox.id;
        checkboxId.push(targetId); // store checked ids
      }
    });
    // eslint-disable-next-line  radix
    checkboxId.forEach((checkbox) => counter.push(parseInt(checkbox)));
    const checked = tasks.filter((task) => !counter.includes(task.index)); // filter ids not checked
    checked.forEach((task, index) => { // set new index
      task.index = index;
      return task.index;
    });
    localStorage.setItem('tasks', JSON.stringify(checked));
    checkboxId = []; // remove the stored checked ids
    Actions.showTask();
  }

**Review by m05e5 — APPROVED** [review]
Hi @sebalp100 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Optional suggestions

- [ Optional ] You did a nice job and your methods are working but for the clearButton, it would have been much easier and better practice to use `filter`

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**